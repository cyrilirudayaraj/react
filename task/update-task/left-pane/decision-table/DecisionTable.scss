@import '~@athena/forge/sass/forge-abstracts';
.task-decision-table {
  .dt-header {
    border-bottom: 1px;
    border-color: hsl(0, 0%, 59%);
    border-bottom-style: solid;
    display: flex;
    justify-content: space-between;
  }
  .fe_c_accordion-item__header {
    height: 52px;
  }
  .delete_dt {
    display: none;
    height: 20px;
    width: 20px;
  }
  .expand-all-button {
    margin-right: map-get($global-margin, 'large');
  }
  .dt-details {
    padding-bottom: map-get($global-padding, 'large');
  }

  .dt-url {
    padding-top: map-get($global-padding, 'large');
  }

  .dt-url-text {
    padding-top: 3px;
    padding-left: 5px;
  }

  .dt-url a {
    display: inline-flex;
  }

  .dt-url a:hover,
  .dt-url a:visited,
  .dt-url a:focus,
  .dt-url a:link {
    box-shadow: none;
    color: none;
    border: none;
    transition: none;
    outline: none;
  }

  .ref-url .fe_l_grid-row__col {
    min-width: 100%;
    padding-right: 0px;
  }

  .branch-url .fe_l_grid-row__col {
    min-width: 100%;
    padding-left: 0px;
  }

  .fe_l_grid-row__col {
    padding-right: 0px;
  }

  .fe_is-expanded,
  .fe_c_accordion-item__header:hover {
    .delete_dt {
      display: block;
    }
  }
  .padding-bottom {
    padding-bottom: 0 !important;
  }

  .margin-top-small {
    margin-top: 10px;
  }
}
.no-records-txt {
  padding: 10px 0px;
}
