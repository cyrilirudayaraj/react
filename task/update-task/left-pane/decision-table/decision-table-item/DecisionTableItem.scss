@import '~@athena/forge/sass/forge-abstracts';
.task-dt {
  .delete-icon {
    .fe_c_icon__icon--primary {
      fill: #d32e45;
    }
  }

  .dt-url {
    padding-top: map-get($global-padding, 'large');
  }

  .dt-url-text {
    padding-top: 5px;
    padding-left: 5px;
  }

  .dt-url a {
    display: inline-flex;
  }

  .dt-url a:hover,
  .dt-url a:visited,
  .dt-url a:focus,
  .dt-url a:link {
    box-shadow: none;
    color: none;
    border: none;
    transition: none;
    outline: none;
  }

  .padding-bottom {
    padding-bottom: 0 !important;
  }

  .max-width .fe_l_grid-row__col {
    min-width: 100%;
  }

  .task-dt .fe_c_form-field {
    padding-bottom: 0 !important;
  }
}
.hide {
  display: none !important;
}
