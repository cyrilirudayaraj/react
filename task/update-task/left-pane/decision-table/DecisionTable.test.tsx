import { configure, mount } from 'enzyme';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import DecisionTable from './DecisionTable';
import { Accordion, AccordionItem, Button, Modal } from '@athena/forge';
import { Formik } from 'formik';
import DecisionTableItem from './decision-table-item/DecisionTableItem';
import { TaskDetail } from '../../../../../types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { act } from 'react-dom/test-utils';

configure({ adapter: new Adapter() });

const taskDetails = {} as TaskDetail;
taskDetails.activeTaskStepId = '1';
const store = configureStore([])({
  task: {
    taskDetails: taskDetails,
  },
  masterData: { users: [] },
  userPermission: {
    userPermissions: ['task.update'],
  },
  updateTaskDetails: jest.fn(),
});
store.dispatch = jest.fn();

describe('DecisionTable', () => {
  let wrapper: any;
  const props: any = {
    taskDecisionTableDetails: [
      {
        description: '<p></p>â†µ',
        id: '5571',
        lastModified: null,
        lastModifiedBy: null,
        modelName: 'Test',
        name: 'Test',
        ordering: null,
        refId: '82',
        refUrl: 'www.test.com',
        taskId: '2093',
        version: '1',
      },
    ],
  };
  const mandatoryFields: any = {
    description: true,
    modelDesignChanges: true,
    name: true,
    testClaimExample: true,
  };

  beforeEach(() => {
    let outerProps;
    const { taskDecisionTableDetails } = props;

    wrapper = mount(
      <Provider store={store}>
        <Formik
          initialValues={{
            taskDecisionTableDetails,
          }}
          onSubmit={jest.fn()}
          validate={jest.fn()}
        >
          {(formik) => {
            outerProps = {
              formik,
              mandatoryFields,
              onSave: jest.fn(),
              onReturn: jest.fn(),
            };
            return <DecisionTable {...outerProps} />;
          }}
        </Formik>
      </Provider>
    );
  });
  it('should render as expected', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('it should have button', () => {
    expect(wrapper.find(Button)).toHaveLength(3);
  });

  it('it should have Accordion', () => {
    expect(wrapper.find(Accordion)).toHaveLength(1);
  });

  it('it should have AccordionItem', () => {
    expect(wrapper.find(AccordionItem)).toHaveLength(1);
  });

  it('it should have DecisionTableItem', () => {
    expect(wrapper.find(DecisionTableItem)).toHaveLength(1);
  });

  it('it should expand toggle', () => {
    act(() => {
      wrapper.find(Button).first().simulate('click');
    });
  });

  it('it should add and delete DecisionTableItem', async () => {
    await act(async () => {
      wrapper.find('.add-button').last().simulate('click');

      const event = { stopPropagation: jest.fn() };
      wrapper.find(DecisionTableItem).first().props().deleteAction(event);

      expect(wrapper.find(Modal)).toHaveLength(1);
      wrapper.find(Modal).props().onPrimaryClick();
    });
  });
});
