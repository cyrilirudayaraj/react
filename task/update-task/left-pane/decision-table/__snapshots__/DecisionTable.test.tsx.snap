// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DecisionTable should render as expected 1`] = `
"<Provider store={{...}}>
  <Formik initialValues={{...}} onSubmit={[Function: mockConstructor]} validate={[Function: mockConstructor]}>
    <DecisionTable formik={{...}} mandatoryFields={{...}} onSave={[Function: mockConstructor]} onReturn={[Function: mockConstructor]}>
      <div className=\\"task-decision-table\\">
        <ThemeConsumerHOC show={false} headerText=\\"Delete\\" alertType=\\"attention\\" width=\\"small\\" onHide={[Function: hidePopup]} onPrimaryClick={[Function: deleteDecisionTableDetails]} primaryButtonText=\\"Delete\\">
          <Modal show={false} headerText=\\"Delete\\" alertType=\\"attention\\" width=\\"small\\" onHide={[Function: hidePopup]} onPrimaryClick={[Function: deleteDecisionTableDetails]} primaryButtonText=\\"Delete\\" theme={{...}} closeButtonAriaLabel=\\"Close Modal\\" disableClose={false} includeButtonRow={true} padded={true} secondaryButtonText=\\"Cancel\\">
            <ModalManagerWrapper onHide={[Function: bound handleHide]} show={false} dialogClass={{...}} className=\\"fe_c_modal fe_c_modal--small fe_is-attention fe_f_all\\">
              <Overlay manager={{...}} onHide={[Function: bound handleHide]} show={false} dialogClass={{...}} className=\\"fe_c_modal fe_c_modal--small fe_is-attention fe_f_all\\" autoFocus={false} enforceFocus={false}>
                <Modal aria-modal=\\"true\\" aria-labelledby={[undefined]} aria-describedby={[undefined]} renderBackdrop={[Function: renderBackdrop]} transition={[Function: FadeUp]} backdropTransition={[Function: Fade]} manager={{...}} onShow={[Function (anonymous)]} onHide={[Function: bound handleHide]} show={false} autoFocus={false} enforceFocus={false} className=\\"fe_c_overlay fe_f_all fe_c_modal fe_c_modal--small fe_is-attention fe_f_all\\" role=\\"dialog\\" backdrop={true} keyboard={true} restoreFocus={true} />
              </Overlay>
            </ModalManagerWrapper>
          </Modal>
        </ThemeConsumerHOC>
        <div className=\\"dt-details\\">
          <FormikConnect(FieldArrayInner) name=\\"taskDecisionTableDetails\\" render={[Function: render]}>
            <FieldArrayInner name=\\"taskDecisionTableDetails\\" render={[Function: render]} formik={{...}} validateOnChange={true}>
              <div>
                <div className=\\"dt-header\\">
                  <div>
                    <Heading headingTag=\\"h3\\" variant=\\"subsection\\" text=\\"Decision Table Changes\\">
                      <h3 className=\\"fe_c_heading fe_c_heading--subsection\\">
                        Decision Table Changes
                      </h3>
                    </Heading>
                  </div>
                  <div>
                    <ForwardRef className=\\"expand-all-button\\" text=\\"Expand All\\" variant=\\"tertiary\\" onClick={[Function: onClick]} disabled={false} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                      <button className=\\"fe_c_button fe_c_button--tertiary fe_c_button--medium expand-all-button\\" aria-disabled={false} aria-label={[undefined]} onClick={[Function: handleClick]} type=\\"button\\">
                        <div className=\\"fe_c_button__content\\">
                          <span className=\\"fe_c_button__text fe_c_button__text--medium fe_c_button__text--tertiary\\">
                            Expand All
                          </span>
                        </div>
                      </button>
                    </ForwardRef>
                    <Connect(WFEnableForPermission) permission=\\"task.decision-table-changes.update\\">
                      <WFEnableForPermission permission=\\"task.decision-table-changes.update\\" userPermissions={{...}} dispatch={[Function: mockConstructor]}>
                        <ForwardRef className=\\"add-button\\" text=\\"Add\\" icon=\\"Add\\" variant=\\"tertiary\\" onClick={[Function: onClick]} disabled={false} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                          <button className=\\"fe_c_button fe_c_button--tertiary fe_c_button--medium add-button\\" aria-disabled={false} aria-label={[undefined]} onClick={[Function: handleClick]} type=\\"button\\">
                            <div className=\\"fe_c_button__content\\">
                              <Add title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\">
                                <IconBase viewBox=\\"0 0 25 25\\" title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\">
                                  <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon fe_c_button__icon fe_c_button__icon--with-text fe_c_button__icon--tertiary\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                                    <g>
                                      <g className=\\"fe_c_icon__background\\">
                                        <rect width=\\"25\\" height=\\"25\\" />
                                      </g>
                                      <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                        <g>
                                          <polygon points=\\"13.3 7.8 11.8 7.8 11.8 11.8 7.8 11.8 7.8 13.3 11.8 13.3 11.8 17.2 13.3 17.2 13.3 13.3 17.2 13.3 17.2 11.8 13.3 11.8 13.3 7.8\\" />
                                          <path d=\\"M20,5a11,11,0,0,0-7.5-3.1A11,11,0,0,0,5,5,10.7,10.7,0,0,0,5,20a11,11,0,0,0,7.5,3.1A11,11,0,0,0,20,20,10.7,10.7,0,0,0,20,5ZM18.9,18.9a9.1,9.1,0,0,1-6.4,2.7,9.1,9.1,0,0,1,0-18.2,9.1,9.1,0,0,1,6.4,15.5Z\\" />
                                        </g>
                                      </g>
                                    </g>
                                  </svg>
                                </IconBase>
                              </Add>
                              <span className=\\"fe_c_button__text fe_c_button__text--medium fe_c_button__text--tertiary\\">
                                Add
                              </span>
                            </div>
                          </button>
                        </ForwardRef>
                      </WFEnableForPermission>
                    </Connect(WFEnableForPermission)>
                  </div>
                </div>
                <div>
                  <DecisionTableItem deleted={false} index={0} formik={{...}} mandatoryFields={{...}} onSave={[Function: mockConstructor]} onReturn={[Function: mockConstructor]} expanded={false} deleteAction={[Function: deleteAction]}>
                    <Accordion className=\\"task-dt\\">
                      <List className=\\"fe_c_accordion task-dt\\" padded={false} dividers={false}>
                        <ul className=\\"fe_c_list fe_c_accordion task-dt\\">
                          <AccordionItem onClick={[Function: handleAccordionClick]} headingText=\\"DT-000082\\" headerSlot={{...}} expanded={false} defaultExpanded={false} headingTag=\\"h3\\" mountedWhileHidden={false} padded={true} tabIndex={0}>
                            <ListItem className=\\"fe_c_accordion-item\\" onClick={[Function: handleAccordionClick]}>
                              <li className=\\"fe_c_list__item fe_c_accordion-item\\" onClick={[Function: handleAccordionClick]}>
                                <div className=\\"fe_c_accordion-item__header\\" onClick={[Function: bound toggleExpand]} onKeyDown={[Function: onKeyDown]} aria-expanded={false} role=\\"button\\" tabIndex={0}>
                                  <Icon icon=\\"Expand\\" title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\" iconSet={{...}}>
                                    <Expand title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\">
                                      <IconBase viewBox=\\"0 0 25 25\\" title=\\"Expand\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_accordion-item__expand\\">
                                        <svg xmlns=\\"http://www.w3.org/2000/svg\\" aria-hidden=\\"true\\" focusable=\\"false\\" className=\\"fe_c_icon fe_c_accordion-item__expand\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                                          <title>
                                            Expand
                                          </title>
                                          <g>
                                            <g className=\\"fe_c_icon__background\\">
                                              <rect width=\\"25\\" height=\\"25\\" />
                                            </g>
                                            <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                              <polygon points=\\"12.5 16.6 6 10 7 9 12.5 14.4 18 9 19 10 12.5 16.6\\" />
                                            </g>
                                          </g>
                                        </svg>
                                      </IconBase>
                                    </Expand>
                                  </Icon>
                                  <Heading text=\\"DT-000082\\" variant=\\"section\\" headingTag=\\"h3\\" className=\\"fe_c_accordion-item__heading-text\\">
                                    <h3 className=\\"fe_c_heading fe_c_heading--section fe_c_accordion-item__heading-text\\">
                                      DT-000082
                                    </h3>
                                  </Heading>
                                  <div className=\\"fe_c_accordion-item__header-slot\\">
                                    <div className=\\"delete_dt\\">
                                      <Connect(WFEnableForPermission) permission=\\"task.decision-table-changes.update\\">
                                        <WFEnableForPermission permission=\\"task.decision-table-changes.update\\" userPermissions={{...}} dispatch={[Function: mockConstructor]}>
                                          <ForwardRef className=\\"delete-icon\\" onClick={[Function: onClick]} icon=\\"Delete\\" variant=\\"tertiary\\" disabled={false} fullWidth={false} iconSet={{...}} size=\\"medium\\" showLoadingSpinner={false} type=\\"button\\" useLink={false}>
                                            <button className=\\"fe_c_button fe_c_button--tertiary fe_c_button--medium delete-icon\\" aria-disabled={false} aria-label=\\"Delete\\" onClick={[Function: handleClick]} type=\\"button\\">
                                              <div className=\\"fe_c_button__content\\">
                                                <Delete title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--tertiary\\">
                                                  <IconBase viewBox=\\"0 0 25 25\\" title={{...}} disabled={false} className=\\"fe_c_button__icon fe_c_button__icon--tertiary\\">
                                                    <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon fe_c_button__icon fe_c_button__icon--tertiary\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                                                      <g>
                                                        <g className=\\"fe_c_icon__background\\">
                                                          <rect width=\\"25\\" height=\\"25\\" />
                                                        </g>
                                                        <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                                          <g>
                                                            <path d=\\"M22.2,4.4H16.9V3.8A1.9,1.9,0,0,0,15,2H10A1.9,1.9,0,0,0,8.1,3.8v.6H2.8V5.9H4.4V23H17.2a3.4,3.4,0,0,0,3.4-3.3V5.9h1.6ZM19.1,19.7a1.9,1.9,0,0,1-1.9,1.8H5.9V5.9H19.1ZM9.6,3.8c0-.2.2-.3.4-.3h5c.2,0,.4.1.4.3v.6H9.6Z\\" />
                                                            <rect x=\\"7.9\\" y=\\"8.6\\" width=\\"1.5\\" height=\\"10.76\\" />
                                                            <rect x=\\"11.8\\" y=\\"8.6\\" width=\\"1.5\\" height=\\"10.76\\" />
                                                            <rect x=\\"15.6\\" y=\\"8.6\\" width=\\"1.5\\" height=\\"10.76\\" />
                                                          </g>
                                                        </g>
                                                      </g>
                                                    </svg>
                                                  </IconBase>
                                                </Delete>
                                              </div>
                                            </button>
                                          </ForwardRef>
                                        </WFEnableForPermission>
                                      </Connect(WFEnableForPermission)>
                                    </div>
                                  </div>
                                </div>
                                <div className=\\"fe_c_accordion-item__content fe_is-exited\\">
                                  <CSSTransition in={false} classNames=\\"fe_is-expanding\\" timeout={400} onEnter={[Function: bound handleEnter]} onEntering={[Function: handleEntering]} onEntered={[Function: handleEntered]} onExit={[Function: handleExit]} onExiting={[Function: handleExiting]} onExited={[Function: bound handleExited]} mountOnEnter={true} unmountOnExit={true}>
                                    <Transition in={false} timeout={400} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} onExited={[Function (anonymous)]} mountOnEnter={true} unmountOnExit={true} appear={false} enter={true} exit={true} />
                                  </CSSTransition>
                                </div>
                              </li>
                            </ListItem>
                          </AccordionItem>
                        </ul>
                      </List>
                    </Accordion>
                  </DecisionTableItem>
                </div>
              </div>
            </FieldArrayInner>
          </FormikConnect(FieldArrayInner)>
        </div>
        <div>
          <GridRow>
            <div className=\\"fe_l_grid-row\\">
              <GridCol width={{...}}>
                <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-10 fe_l_grid-row__col--medium-10 fe_l_grid-row__col--large-10\\">
                  <Connect(WFEnableForPermission) permission=\\"task.model-branch-url.update\\">
                    <WFEnableForPermission permission=\\"task.model-branch-url.update\\" userPermissions={{...}} dispatch={[Function: mockConstructor]}>
                      <FormField maxlength=\\"500\\" labelText=\\"DFM Branch Link\\" labelWidth={1} name=\\"modelBranchUrl\\" value={[undefined]} onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} required={false} autoComplete=\\"off\\" error={[undefined]} id=\\"modelBranchUrl\\" disabled={false} className=\\"branch-url padding-bottom\\" inputAs={{...}}>
                        <FormFieldLayout labelWidth={1} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"branch-url padding-bottom\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
                          <div className=\\"fe_c_form-field fe_c_form-field--medium branch-url padding-bottom\\">
                            <GridRow nested={true}>
                              <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                <GridCol width={{...}} className=\\"fe_c_form-field__label\\">
                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-1 fe_c_form-field__label\\">
                                    <Label id=\\"modelBranchUrl-label\\" htmlFor=\\"modelBranchUrl\\" text=\\"DFM Branch Link\\" disabled={false} required={false} useLegend={false}>
                                      <label htmlFor=\\"modelBranchUrl\\" id=\\"modelBranchUrl-label\\" className=\\"fe_c_label\\">
                                        DFM Branch Link
                                      </label>
                                    </Label>
                                  </div>
                                </GridCol>
                                <GridCol width={{...}} className=\\"fe_c_form-field__right\\">
                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-11 fe_c_form-field__right\\">
                                    <div className=\\"fe_c_form-field__input-slot\\">
                                      <Input maxlength=\\"500\\" value={[undefined]} onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} autoComplete=\\"off\\" className=\\"fe_c_form-field__input\\" id=\\"modelBranchUrl\\" name=\\"modelBranchUrl\\" required={false} hideRequiredStyles={false} disabled={false} error={[undefined]} aria-labelledby=\\"modelBranchUrl-label\\" type=\\"text\\">
                                        <div className=\\"fe_c_input fe_c_form-field__input\\">
                                          <input maxlength=\\"500\\" value={[undefined]} onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} autoComplete=\\"off\\" id=\\"modelBranchUrl\\" name=\\"modelBranchUrl\\" aria-labelledby=\\"modelBranchUrl-label\\" className=\\"fe_c_input__input\\" type=\\"text\\" disabled={false} required={false} />
                                        </div>
                                      </Input>
                                    </div>
                                    <div className=\\"fe_c_form-field__status\\" />
                                  </div>
                                </GridCol>
                              </div>
                            </GridRow>
                          </div>
                        </FormFieldLayout>
                      </FormField>
                    </WFEnableForPermission>
                  </Connect(WFEnableForPermission)>
                </div>
              </GridCol>
              <GridCol width={{...}}>
                <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-2 fe_l_grid-row__col--medium-2 fe_l_grid-row__col--large-2\\">
                  <div className=\\"dt-url padding-bottom\\">
                    <a href=\\"//\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
                      <div>
                        <Icon icon=\\"NavigateAway\\" iconSet={{...}}>
                          <NavigateAway title=\\"Navigate Away\\">
                            <IconBase viewBox=\\"0 0 25 25\\" title=\\"Navigate Away\\" className={[undefined]}>
                              <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                                <title>
                                  Navigate Away
                                </title>
                                <g>
                                  <g className=\\"fe_c_icon__background\\">
                                    <rect width=\\"25\\" height=\\"25\\" />
                                  </g>
                                  <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                    <g>
                                      <path d=\\"M20,19.2a2.3,2.3,0,0,1-2.3,2.3H7V3.5H20V6.1h1.5V2H5.5V23H17.7a3.8,3.8,0,0,0,3.8-3.8v-.3H20Z\\" />
                                      <polygon points=\\"18.6 17.9 24 12.5 18.6 7.1 17.5 8.2 21.1 11.7 13.5 11.7 13.5 13.2 21.1 13.2 17.5 16.8 18.6 17.9\\" />
                                    </g>
                                  </g>
                                </g>
                              </svg>
                            </IconBase>
                          </NavigateAway>
                        </Icon>
                      </div>
                      <div className=\\"dt-url-text\\">
                        <span>
                          Go
                        </span>
                      </div>
                    </a>
                  </div>
                </div>
              </GridCol>
            </div>
          </GridRow>
          <GridRow>
            <div className=\\"fe_l_grid-row\\">
              <GridCol width={{...}}>
                <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-10 fe_l_grid-row__col--medium-10 fe_l_grid-row__col--large-10\\">
                  <Connect(WFEnableForPermission) permission=\\"task.dt-branch-url.update\\">
                    <WFEnableForPermission permission=\\"task.dt-branch-url.update\\" userPermissions={{...}} dispatch={[Function: mockConstructor]}>
                      <FormField maxlength=\\"500\\" labelText=\\"ODM Branch Link\\" labelWidth={1} name=\\"dtBranchUrl\\" value={[undefined]} onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} required={false} autoComplete=\\"off\\" error={[undefined]} id=\\"dtBranchUrl\\" disabled={false} className=\\"branch-url padding-bottom\\" inputAs={{...}}>
                        <FormFieldLayout labelWidth={1} nested={[undefined]} layout={[undefined]} statusExtras={[undefined]} className=\\"branch-url padding-bottom\\" labelAlwaysAbove={[undefined]} labelSlot={{...}} messageSlot={[undefined]} statusIndicatorSlot={[undefined]} inputSlot={{...}}>
                          <div className=\\"fe_c_form-field fe_c_form-field--medium branch-url padding-bottom\\">
                            <GridRow nested={true}>
                              <div className=\\"fe_l_grid-row fe_l_grid-row--nested\\">
                                <GridCol width={{...}} className=\\"fe_c_form-field__label\\">
                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-1 fe_c_form-field__label\\">
                                    <Label id=\\"dtBranchUrl-label\\" htmlFor=\\"dtBranchUrl\\" text=\\"ODM Branch Link\\" disabled={false} required={false} useLegend={false}>
                                      <label htmlFor=\\"dtBranchUrl\\" id=\\"dtBranchUrl-label\\" className=\\"fe_c_label\\">
                                        ODM Branch Link
                                      </label>
                                    </Label>
                                  </div>
                                </GridCol>
                                <GridCol width={{...}} className=\\"fe_c_form-field__right\\">
                                  <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-12 fe_l_grid-row__col--medium-11 fe_c_form-field__right\\">
                                    <div className=\\"fe_c_form-field__input-slot\\">
                                      <Input maxlength=\\"500\\" value={[undefined]} onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} autoComplete=\\"off\\" className=\\"fe_c_form-field__input\\" id=\\"dtBranchUrl\\" name=\\"dtBranchUrl\\" required={false} hideRequiredStyles={false} disabled={false} error={[undefined]} aria-labelledby=\\"dtBranchUrl-label\\" type=\\"text\\">
                                        <div className=\\"fe_c_input fe_c_form-field__input\\">
                                          <input maxlength=\\"500\\" value={[undefined]} onChange={[Function (anonymous)]} onBlur={[Function (anonymous)]} autoComplete=\\"off\\" id=\\"dtBranchUrl\\" name=\\"dtBranchUrl\\" aria-labelledby=\\"dtBranchUrl-label\\" className=\\"fe_c_input__input\\" type=\\"text\\" disabled={false} required={false} />
                                        </div>
                                      </Input>
                                    </div>
                                    <div className=\\"fe_c_form-field__status\\" />
                                  </div>
                                </GridCol>
                              </div>
                            </GridRow>
                          </div>
                        </FormFieldLayout>
                      </FormField>
                    </WFEnableForPermission>
                  </Connect(WFEnableForPermission)>
                </div>
              </GridCol>
              <GridCol width={{...}}>
                <div className=\\"fe_l_grid-row__col fe_l_grid-row__col--small-2 fe_l_grid-row__col--medium-2 fe_l_grid-row__col--large-2\\">
                  <div className=\\"dt-url padding-bottom\\">
                    <a href=\\"//\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">
                      <div>
                        <Icon icon=\\"NavigateAway\\" iconSet={{...}}>
                          <NavigateAway title=\\"Navigate Away\\">
                            <IconBase viewBox=\\"0 0 25 25\\" title=\\"Navigate Away\\" className={[undefined]}>
                              <svg xmlns=\\"http://www.w3.org/2000/svg\\" className=\\"fe_c_icon\\" viewBox=\\"0 0 25 25\\" style={{...}}>
                                <title>
                                  Navigate Away
                                </title>
                                <g>
                                  <g className=\\"fe_c_icon__background\\">
                                    <rect width=\\"25\\" height=\\"25\\" />
                                  </g>
                                  <g className=\\"fe_c_icon__icon fe_c_icon__icon--primary\\">
                                    <g>
                                      <path d=\\"M20,19.2a2.3,2.3,0,0,1-2.3,2.3H7V3.5H20V6.1h1.5V2H5.5V23H17.7a3.8,3.8,0,0,0,3.8-3.8v-.3H20Z\\" />
                                      <polygon points=\\"18.6 17.9 24 12.5 18.6 7.1 17.5 8.2 21.1 11.7 13.5 11.7 13.5 13.2 21.1 13.2 17.5 16.8 18.6 17.9\\" />
                                    </g>
                                  </g>
                                </g>
                              </svg>
                            </IconBase>
                          </NavigateAway>
                        </Icon>
                      </div>
                      <div className=\\"dt-url-text\\">
                        <span>
                          Go
                        </span>
                      </div>
                    </a>
                  </div>
                </div>
              </GridCol>
            </div>
          </GridRow>
        </div>
      </div>
    </DecisionTable>
  </Formik>
</Provider>"
`;
